addon/background.ts

```diff
@@ -10,18 +11,79 @@ const autoconsentConfig: Config = {
   detectRetries: 20,
 };
 
-async function loadRules(): Promise<RuleBundle> {
+// Storage abstraction: MV2 keeps everything in memory, MV3 uses chrome.storage
+const storage: {[key: string]: any} = {};
+async function storageSet(obj: {[key: string]: any}): Promise<void> {
+  if (manifestVersion === 2) {
+    Object.assign(storage, obj);
+    return;
+  } 
+  return chrome.storage.local.set(obj);
+}
+
+async function storageGet(key: string): Promise<{[key: string]: any}> {
+  if (manifestVersion === 2) {
+    return storage;
+  }
+  return chrome.storage.local.get(key);
+}
+
+async function storageRemove(key: string): Promise<void> {
+  if (manifestVersion === 2) {
+    delete storage[key];
+    return;
+  }
+  return chrome.storage.local.remove(key);
+}
+
+async function loadRules() {
   const res = await fetch("./rules.json");
-  return await res.json();
+  storageSet({
+    rules: await res.json(),
+  });
 }
 
-chrome.runtime.onInstalled.addListener(() => {
-  loadRules().then(rules => {
-    chrome.storage.local.set({
-      rules: rules,
+async function evalInTab(tabId: number, frameId: number, code: string): Promise<chrome.scripting.InjectionResult[]> {
+  if (manifestVersion === 2) {
+    return new Promise((resolve) => {
+      chrome.tabs.executeScript(tabId, {
+        frameId,
+        code: `window.eval(JSON.parse('${JSON.stringify(code)}'))`
+      }, (result) => {
+        resolve([{
+          result,
+          frameId,
+        }]);
+      })
     });
-  });
+  }
+  return chrome.scripting.executeScript({
+    target: {
+      tabId,
+      frameIds: [frameId],
+    },
+    world: "MAIN",
+    args: [code],
+    func: (code) => {
+      try {
+        return window.eval(code);
+      } catch (e) {
+        // ignore CSP errors
+        return;
+      }
+    },
+  })
+}
+
+chrome.runtime.onStartup.addListener(() => {
+  loadRules()
 });
+if (manifestVersion === 2) {
+  // always load rules on startup in MV2
+  loadRules()
+}
+
+(<any>window).debug = storage;
```

Comment:
This breaks the Chrome extension
<img width="679" alt="image" src="https://user-images.githubusercontent.com/2726132/179471976-a4657fb0-d0c5-4cda-a22a-5d39ce5e4c96.png">

Reply:
Sorry - debug code I forgot to remove.